### ISSUE TITLE
Logout from custom oauth does not log user out

### ISSUE THREAD

**Issue by xtilyn** [RocketChat/Rocket.Chat.ReactNative#1818]
- Your Rocket.Chat Experimental app version: 4.4.0
- Your Rocket.Chat server version: 3.0.0
- Device (or Simulator) you're running with: Android (9) and IOS (11.4)

When authenticated through custom oauth, logging out from the server does not completely log user out. 

Steps to reproduce:
- Authenticate through custom oauth
- Logout
- Log back in --> app redirects back in to the previous signed in account without user having to enter a new credentials.

**Comment by akumar1999** [comment]
xtilyn,
I did some looking up and found that this is somewhat standard oAuth behavior; as long as you are only logging in from Rocket.Chat, but not the OAuth service (Google, for example), you will never really "log out". 

Are you having the problem despite logging out from Google on your phone? In that case I understand how this could be a problem; seems like if I login using Google account 'x', I could be stuck with that account unless I do a complete reset of the app.

**Comment by xtilyn** [comment]
Hi akumar1999,
My company has implemented their own SSO service with OpenID Connect. The main issue is when a user wants to logout and sign in to a different account, current Rocket Chat does not allow them to do so. Unless we clear app storage, rocket chat will just redirect them back to the old signed in account. I don't really see this as an expected behavior but then again, I'm not that familiar with the oAuth framework. Is there a possibility that it could have something to do with our custom oauth service?

**Comment by diegolmello** [comment]
Most of the users may want to keep connected to their 3rd party login services instead of having to enter credentials over and over.
I commented how this could be achieved on your PR.
I'm closing this issue, but feel free to keep the discussion below.
Thanks.

**Comment by xtilyn** [comment]
When a user logs in, their session is still kept once they come back to the app (unless their token expires), so they don't have to enter their credentials over and over. But when they want to logout within the lifetime of their token, they can't. I'm not sure how this could be solved by making webview incognito but I will try it out. Thanks for the response!

**Comment by akumar1999** [comment]
Hi @diegolmello, I understand your point that users does not like entering their credentials over and over again. However, while playing around with the OAuth in rocket chat and another 3rd party app (using google for example), I realized that when I logout of rocket chat and when I logout from my google account from my phone which I used to log in, the app does not ask you to re-enter your credentials, in fact, the user data is saved in the rocket chat app.   Whereas if I log in using Oauth (google for example) in other 3rd party apps and when I log out from the app and my google account from my phone, the apps do ask me to re-enter the credentials in order to login to the app. So, I feel this is a small issue that we should look into.
